lib/_http_outgoing.js

```diff
@@ -448,9 +449,23 @@ function _storeHeader(firstLine, headers) {
         (state.contLen || this.useChunkedEncodingByDefault || this.agent);
     if (shouldSendKeepAlive) {
       header += 'Connection: keep-alive' + CRLF;
```

Comment:
If this is the last request of the socket, we should be setting this to `'Connection: close'`.

Reply:
If this is the last request this code will not be executed, the `connection: close` is set here
https://github.com/nodejs/node/blob/master/lib/_http_outgoing.js#L457

because I set `res.shouldKeepAlive = false` here
https://github.com/nodejs/node/pull/40082/files/64132000c63934af235ae59095faad2015b3a045#diff-2464899ab38c61dd42654fdb9658c75989cb6883b9c9d76613872a12d3fe2780R918